File: src/Web/aspnet-core/src/Piggyvault.Application/Piggy/Transactions/Dto/TransactionEditDto.cs
using System.ComponentModel.DataAnnotations;
using Abp.AutoMapper;
using System;

namespace Piggyvault.Piggy.Transactions.Dto
{
    [AutoMap(typeof(Transaction))]
    public class TransactionEditDto
    {
        public virtual Guid AccountId { get; set; }
        public virtual decimal Amount { get; set; }
        public virtual Guid CategoryId { get; set; }

        [MaxLength(Transaction.MaxDescriptionLength)]
        public virtual string Description { get; set; }

        public virtual Guid? Id { get; set; }

        [Required]
        public virtual DateTime TransactionTime { get; set; }
    }
}

File: src/Mobile/piggy_flutter/lib/models/transaction_edit_dto.dart
class TransactionEditDto {
  String? id, description, accountId, transactionTime;
  double? amount;
  String? categoryId;

  TransactionEditDto(
      {this.id,
      this.description,
      this.accountId,
      this.transactionTime,
      this.amount,
      this.categoryId});

  TransactionEditDto.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        description = json['description'],
        amount = json['amount'],
        categoryId = json['categoryId'],
        transactionTime = json['transactionTime'],
        accountId = json['accountId'];
}
